catroid/src/main/java/org/catrobat/catroid/content/actions/RepeatUntilAction.kt

```diff
@@ -0,0 +1,95 @@
+/*
+ * Catroid: An on-device visual programming system for Android devices
+ * Copyright (C) 2010-2018 The Catrobat Team
+ * (<http://developer.catrobat.org/credits>)
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * An additional term exception under section 7 of the GNU Affero
+ * General Public License, version 3, is available at
+ * http://developer.catrobat.org/license_additional_term
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.catrobat.catroid.content.actions
+
+import android.util.Log
+import com.badlogic.gdx.scenes.scene2d.actions.RepeatAction
+import org.catrobat.catroid.content.Sprite
+import org.catrobat.catroid.formulaeditor.Formula
+import org.catrobat.catroid.formulaeditor.InterpretationException
+
+class RepeatUntilAction : RepeatAction() {
+    var executedCount = 0
+        private set
+    var sprite: Sprite? = null
+    var repeatCondition: Formula? = null
+    private var isCurrentLoopInitialized = false
+    private var currentTime = 0f
+    private val isValidConditionFormula: Boolean
+        get() {
+            try {
+                repeatCondition?.interpretDouble(sprite) ?: return false
+            } catch (interpretationException: InterpretationException) {
+                Log.d(
+                    javaClass.simpleName, "Formula interpretation for this specific Brick failed.",
+                    interpretationException
+                )
+                return false
+            }
+            return true
+        }
+
+    private val isConditionTrue: Boolean
+        get() = try {
+            repeatCondition?.interpretDouble(sprite) != 0.0
+        } catch (interpretationException: InterpretationException) {
+            Log.d(
+                javaClass.simpleName, "Formula interpretation for this specific Brick failed.",
+                interpretationException
+            )
+            true
+        }
+
+    public override fun delegate(delta: Float): Boolean {
+        if (!isValidConditionFormula) {
+            return true
+        }
+        if (!isCurrentLoopInitialized) {
+            if (isConditionTrue) {
+                return true
+            }
+            currentTime = 0f
+            isCurrentLoopInitialized = true
+        }
+        currentTime += delta
+        if (action.act(delta) && currentTime >= LOOP_DELAY) {
+            executedCount++
+            if (isConditionTrue) {
```

Comment:
PLEASE-FIX: refactor the boolean flags to functions, since from line 68 till here, the code implies that the value of **isConditionTrue** might magically change.  Please, only use get()/set() for small code to verify input etc.. 
https://kotlinlang.org/docs/reference/properties.html#backing-fields